src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageDetailsTabViewModel.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public class PackageDetailsTabViewModel : ViewModelBase, IDisposable
+    {
+        private bool _disposed = false;
+
+        private bool _readmeTabEnabled;
+
+        public ReadmePreviewViewModel ReadmePreviewViewModel { get; private set; }
+
+        public DetailControlModel DetailControlModel { get; private set; }
+
+        public ObservableCollection<RenderedViewModelBase> Tabs { get; private set; }
+
+        private RenderedViewModelBase _selectedTab;
+        public RenderedViewModelBase SelectedTab
+        {
+            get => _selectedTab;
+            set
+            {
+                SetAndRaisePropertyChanged(ref _selectedTab, value);
+            }
+        }
+
+        public PackageDetailsTabViewModel()
+        {
+            _readmeTabEnabled = true;
+            Tabs = new ObservableCollection<RenderedViewModelBase>();
+        }
+
+        public PackageMetadataTab GetSelectedTab()
+        {
+            return SelectedTab is DetailControlModel ? PackageMetadataTab.PackageDetails : PackageMetadataTab.Readme;
+        }
+
+        public async Task InitializeAsync(DetailControlModel detailControlModel, INuGetPackageFileService nugetPackageFileService, ItemFilter currentFilter, PackageMetadataTab initialSelectedTab)
+        {
+            var nuGetFeatureFlagService = await ServiceLocator.GetComponentModelServiceAsync<INuGetFeatureFlagService>();
+            _readmeTabEnabled = await nuGetFeatureFlagService.IsFeatureEnabledAsync(NuGetFeatureFlagConstants.RenderReadmeInPMUI);
+
+            ReadmePreviewViewModel = new ReadmePreviewViewModel(nugetPackageFileService, currentFilter);
+            ReadmePreviewViewModel.IsVisible = _readmeTabEnabled;
+            DetailControlModel = detailControlModel;
+
+            Tabs.Add(ReadmePreviewViewModel);
+            Tabs.Add(DetailControlModel);
+
+            SelectedTab = _readmeTabEnabled && initialSelectedTab == PackageMetadataTab.Readme
+                ? ReadmePreviewViewModel
+                : DetailControlModel;
```

Comment:
I left this comment on a previous iteration but didn't submit my review before this latest commit so I'll leave it here:

This works fine for a 2-tab system. If we want to make sure this is resilient for future changes that has more than 2 tabs, we can iterate over the list of tabs to find the first one that's visible and set it to the SelectedTab, taking into account the `initialSelectedTab` as well.

Reply:
Okay switched it a bit, lemme know what you think